models/reviews_models.js

```diff
@@ -90,10 +95,36 @@ exports.updateReviewById = (body, params) => {
       status: 400,
       msg: "Invalid request: Enter a valid order query",
     });
-  } else {
-    queryString += ` ORDER BY ${sort_by} ${order.toUpperCase()}`;
-  }
-  return db.query(queryString).then(({ rows: reviews }) => {
-    return reviews;
-  });
-  };
\ No newline at end of file
+  } 
+  if(category){
+        queryString += `WHERE category = $1`
+        queryValues.push(category);
+      } 
+      queryString += `GROUP BY reviews.review_id`; 
+      queryString += ` ORDER BY ${sort_by} ${order.toUpperCase()}`;
+      return db.query(queryString, queryValues).then(({ rows: reviews }) => {
+        if (!reviews[0]) {
+          return fetchAllCategories().then((category) => {
+            const allCategories = category.map((category) => {
```

Comment:
Could we not just query the topics table and see if that topic exists? The query could still be inside a function itself. But I think think getting all the categories back and then checking if it is in there, seems a longer way to do this